WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/quickorder/QuickOrderViewModel.kt

```diff
@@ -37,10 +49,42 @@ class QuickOrderViewModel @Inject constructor(
             )
         }
 
+    fun onDoneButtonClicked() {
+        createQuickOrder()
+    }
+
+    fun createQuickOrder() {
+        if (!networkStatus.isConnected()) {
+            triggerEvent(MultiLiveEvent.Event.ShowSnackbar(R.string.offline_error))
+            return
+        }
+
+        viewState = viewState.copy(isProgressShowing = true, isDoneButtonEnabled = false)
+
+        launch {
+            val result = orderStore.postQuickOrder(
+                selectedSite.get(),
+                viewState.currentPrice.toString()
+            )
```

Comment:
we're not shifting the launch context here, so this request will end up getting triggered inside a coroutine of `Dispatchers.Main` already, I suggest we launch it using the `dispatchers.io` through injection of the `CoroutineDispatchers` like this:
```suggestion
               launch(dispatchers.io) {
            val result = orderStore.postQuickOrder(
                selectedSite.get(),
                viewState.currentPrice.toString()
            )
```

Or we could just remove the `withContext(Dispatchers.Main)` for the following code, but we would still be triggering the request inside a Main dispatcher coroutine, which could lead us to ANRs. WDYT?

Reply:
@ThomazFB Thanks for the tip, I'm still wrapping my head around coroutine contexts so any advice is welcome. I made the suggested change in da9dc5a.